nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -105,11 +122,67 @@ async fn run() -> Result<()> {
         while let Some(msg) = relay_channel.recv().await {
             match msg {
                 RelayMessage::Event(ev) => match ev.event.kind {
+                    // Save 6002 event, for later 6003 verify
+                    Kind::ASSIGN => {
+                        let assign_event_id = CryptoHash::new(ev.event.id);
+                        let db_entry = AssignedJobs {
+                            _id: CryptoHash::new(ev.event.id),
+                            assign_event: ev.event.clone(),
+                        };
+                        // Save assign event to db
+                        match assigned_jobs_collection.insert_one(db_entry, None).await {
+                            Ok(_) => {
+                                tracing::debug!(
+                                    "Write assigned job to db: {:?}",
+                                    assign_event_id.hash()
+                                );
+                            }
+                            Err(err) => {
+                                tracing::error!("Cannot write assigned job to db: {}", err);
+                            }
+                        }
+                    }
                     Kind::RESULT => {
                         if let Ok(payload) = get_result_payload(&ev) {
                             if let Ok(Some(job_id)) = get_single_tag_entry('e', &ev.event.tags) {
+                                // Get saved assign event from db
+                                let assigned_job = assigned_jobs_collection
+                                    .find_one(doc! { "_id": job_id.clone() }, None)
+                                    .await
+                                    .unwrap_or(None);
+
+                                if let None = assigned_job {
+                                    tracing::error!("Can't find assign_event from job_id");
+                                    continue;
+                                }
+
+                                let assign_event = assigned_job.unwrap().assign_event;
+
+                                // Check assigner in whitelist
+                                if !assigner_whitelist.contains(&assign_event.sender) {
```

Comment:
you don't need to do this check since you should subscribe events from whitelisted assigners only

Reply:
Then we need to put Result and Assign subscription into two different subscriptions

Reply:
I found I can send multiple filters in a single subscription, this is fixed